tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/FewerBracesSuite.scala

```diff
@@ -990,4 +990,563 @@ class FewerBracesSuite extends BaseDottySuite {
     runTestAssert[Stat](code, Some(layout))(tree)
   }
 
+  test("scalafmt #3720 1 simple, space after op, top-level, not indented") {
+    val code =
+      """
+        |baz:
+        |  arg
+        |++ qux
+        |""".stripMargin
+    val layout = "baz(arg) ++ qux"
+    val tree = Term.ApplyInfix(
+      Term.Apply(tname("baz"), List(tname("arg"))),
+      tname("++"),
+      Nil,
+      List(tname("qux"))
+    )
+    runTestAssert[Stat](code, Some(layout))(tree)
+  }
+
+  test("scalafmt #3720 1 simple, space after op, top-level, indented") {
+    val code =
+      """
+        |  baz:
+        |    arg
+        |  ++ qux
+        |""".stripMargin
+    val layout = "baz(arg ++ qux)"
+    val tree = Term.Apply(
+      tname("baz"),
+      Term.ApplyInfix(tname("arg"), tname("++"), Nil, List(tname("qux"))) :: Nil
+    )
+    runTestAssert[Stat](code, Some(layout))(tree)
+  }
+
+  test("scalafmt #3720 1 simple, space after op, object in braces") {
+    val code =
+      """
+        |object a {
+        |  baz:
+        |    arg
+        |  ++ qux
+        |}
+        |""".stripMargin
+    val layout = "object a { baz(arg) ++ qux }"
+    val tree = Defn.Object(
+      Nil,
+      tname("a"),
+      tpl(
+        Term.ApplyInfix(
+          Term.Apply(tname("baz"), List(tname("arg"))),
+          tname("++"),
+          Nil,
+          List(tname("qux"))
+        ) :: Nil
+      )
+    )
+    runTestAssert[Stat](code, Some(layout))(tree)
+  }
+
+  test("scalafmt #3720 1 simple, space after op, object indented") {
+    val code =
+      """
+        |object a:
+        |  baz:
+        |    arg
+        |  ++ qux
+        |""".stripMargin
+    val layout = "object a { baz(arg) ++ qux }"
+    val tree = Defn.Object(
+      Nil,
+      tname("a"),
+      tpl(
+        Term.ApplyInfix(
+          Term.Apply(tname("baz"), List(tname("arg"))),
+          tname("++"),
+          Nil,
+          List(tname("qux"))
+        ) :: Nil
+      )
+    )
+    runTestAssert[Stat](code, Some(layout))(tree)
+  }
+
+  test("scalafmt #3720 2 complex, break after op, top-level, not indented") {
+    val code =
+      """
+        |foo.bar:
+        |  arg
+        |++
+        |  baz:
+        |    arg
+        |  ++
+        |  qux
+        |""".stripMargin
+    val layout =
+      """
+        |foo.bar(arg) ++ baz {
+        |  arg
+        |  ++
+        |}
+        |""".stripMargin
+    val tree = Term.ApplyInfix(
+      Term.Apply(Term.Select(tname("foo"), tname("bar")), List(tname("arg"))),
+      tname("++"),
+      Nil,
+      Term.Apply(tname("baz"), List(Term.Block(List(tname("arg"), tname("++"))))) :: Nil
+    )
+    runTestAssert[Stat](code, Some(layout))(tree)
+  }
+
+  test("scalafmt #3720 2 complex, break after op, top-level, indented") {
+    val code =
+      """
+        |  foo.bar:
+        |    arg
+        |  ++
+        |  baz:
+        |    arg
+        |  ++
+        |  qux
```

Comment:
Why lost qux in this case?

Reply:
it's a bug. i mentioned it in the commit (and pr) comment:

_However, the fix is incomplete as it doesn't handle the very rare case when the infix expression is at the file level and not nested within a block of any kind. This will be fixed in the next commit._

Reply:
Ach, ok. Sorry about that then!